src/tactic/equiv_rw.lean

```diff
@@ -241,34 +241,68 @@ do
 end tactic
 
 namespace tactic.interactive
-open lean.parser
-open interactive interactive.types
-open tactic
 
-local postfix `?`:9001 := optional
+open tactic
+setup_tactic_parser
+
+/-- Auxiliary function to call `equiv_rw_hyp` on a `list pexpr` recursively. -/
+meta def equiv_rw_hyp_aux (hyp : name) (cfg : equiv_rw_cfg) (permissive : bool := ff) :
+  list expr → itactic
+| []       := skip
+| (e :: t) := do
+  if permissive then equiv_rw_hyp hyp e cfg <|> skip
+  else equiv_rw_hyp hyp e cfg,
+  equiv_rw_hyp_aux t
+
+/-- Auxiliary function to call `equiv_rw_target` on a `list pexpr` recursively. -/
+meta def equiv_rw_target_aux (cfg : equiv_rw_cfg) (permissive : bool) :
+  list expr → itactic
+| []       := skip
+| (e :: t) := do
+  if permissive then equiv_rw_target e cfg <|> skip
+  else equiv_rw_target e cfg,
+  equiv_rw_target_aux t
 
 /--
-`equiv_rw e at h`, where `h : α` is a hypothesis, and `e : α ≃ β`,
-will attempt to transport `h` along `e`, producing a new hypothesis `h : β`,
-with all occurrences of `h` in other hypotheses and the goal replaced with `e.symm h`.
+`equiv_rw e at h₁ h₂ ⋯`, where each `hᵢ : α` is a hypothesis, and `e : α ≃ β`,
+will attempt to transport each `hᵢ` along `e`, producing a new hypothesis `hᵢ : β`,
+with all occurrences of `hᵢ` in other hypotheses and the goal replaced with `e.symm hᵢ`.
 
 `equiv_rw e` will attempt to transport the goal along an equivalence `e : α ≃ β`.
 In its minimal form it replaces the goal `⊢ α` with `⊢ β` by calling `apply e.inv_fun`.
 
-`equiv_rw` will also try rewriting under (equiv_)functors, so can turn
+`equiv_rw e at *` will attempt to apply `equiv_rw e` on the goal and on each expression
+available in the local context (except on `e` itself), failing silently where it can't.
+
+`equiv_rw [e₁, e₂, ⋯] at h₁ h₂ ⋯` is equivalent to
+`{ equiv_rw e₁ at h₁ h₂ ⋯, equiv_rw e₂ at h₁ h₂ ⋯, ⋯ }`.
```

Comment:
I wonder if this is the expected behavior. I believe `rw` instead applies all the rewrites to each of the locations (see `rw_core`, which uses some functions on `location` to apply a tactic to each location). On the other hand, this seems to be how `simp_rw` functions.

Reply:
But it's indeed applying all rewrites to each of the locations

Reply:
What's the current (pre-PR) behavior of `equiv_rw`?

Reply:
@robertylewis It looks like it could only support a single location

Reply:
The pre-PR behavior doesn't accept multiple equivalences nor multiple locations